lib/Sema/CSSimplify.cpp

```diff
@@ -4252,6 +4251,267 @@ ConstraintSystem::simplifyKeyPathApplicationConstraint(
   return unsolved();
 }
 
+/// Returns the function declaration corresponding to a @dynamicCallable
+/// attribute required method (if it exists) implemented by a type. Otherwise,
+/// return nullptr.
+static FuncDecl *
+lookupDynamicCallableMethod(Type type, ConstraintSystem &CS,
+                            const ConstraintLocatorBuilder &locator,
+                            Identifier argumentName, bool hasKeywordArgs,
+                            bool &error) {
```

Comment:
It's not clear to me what value the separate error parameter adds here. Another option might be to have `getDynamicCallableMethods` return an llvm::Optional and use the existence or lack of a value as an indication of the success or failure, unless you want to carry more semantic information in the form of a richer error type than a bool.

Reply:
Thanks for pointing this out. I'll use `llvm::Optional` instead, same for `hasValidDynamicCallableMethod` in `TypeCheckAttr.cpp` which currently also takes a `bool &error` argument.